spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/keras/KerasLayer.scala

```diff
@@ -149,33 +153,52 @@ abstract class KerasLayer[A <: Activity: ClassTag, B <: Activity: ClassTag, T: C
     labor.accGradParameters(input, gradOutput)
   }
 
-  override def isCompatibleWithKeras(): Boolean = true
+  override def isBuilt(): Boolean = {
+    !this.modules.isEmpty && super.isBuilt()
+  }
 
-  override def isCompatibleWithTorch(): Boolean = false
+  override def isKerasStyle(): Boolean = true
 
-  override def getInputShape(): Shape = {
-    if (batchInputShape != null) {
-      batchInputShape
-    } else if (this.labor == null) {
-      null
-    } else {
-      this.labor.getInputShape()
+  override def computeOutputShape(inputShape: Shape): Shape = {
+    val cls = this.getClass.getComponentType
+    val torchLayer = labor match {
+      case s: TContainer[_, _, T] =>
+        s.modules.filter(
+          !_.isInstanceOf[Input[T]]).filter(!_.isInstanceOf[IdentityShapeWrapper[_, _, T]])(0)
+      case l: InferShape => l
     }
+    torchLayer.asInstanceOf[InferShape].computeOutputShape(inputShape)
   }
 
-  override def computeOutputShape(inputShape: Shape): Shape = {
-    this.labor.computeOutputShape(inputShape)
+  private def checkWithCurrentInputShape(calcInputShape: Shape): Unit = {
+    if (getInputShape() != null) {
+      val withoutBatchInputShape = KerasLayer.removeBatch(getInputShape())
+      val withoutBatchCalcInputShape = KerasLayer.removeBatch(calcInputShape)
+      require(withoutBatchInputShape == withoutBatchCalcInputShape,
+        s"InputShape from constructor ${withoutBatchInputShape}" +
+          s"should be the same with the calculated inputShape: ${withoutBatchCalcInputShape}")
+    }
   }
 
-  override def getOutputShape(): Shape = labor.getOutputShape()
+  override def inferShape(calcInputShape: Shape): Shape = {
+    checkWithCurrentInputShape(calcInputShape)
+    super.inferShape(calcInputShape)
+  }
 
-  override def build(inputShape: Shape): Shape = {
-    this.labor = doBuild(inputShape)
-    val outputShape = computeOutputShape(inputShape)
-    this.outputShapeValue ++= Array(outputShape)
-    this.inputShapeValue = inputShape
-    isBuilt = true
-    outputShape // we cannot use getOutputShape here as it may containing multiple value
+  override def build(calcInputShape: Shape): Shape = {
+    this match {
+      case ks: com.intel.analytics.bigdl.nn.keras.Sequential[T] =>
```

Comment:
Why don't we just override build() method in keras.Sequential?

Reply:
would do that same as `Model`